Sally7/RequestExecutor/ConcurrentRequestExecutor.cs

```diff
@@ -92,6 +92,14 @@ public async ValueTask<Memory<byte>> PerformRequest(ReadOnlyMemory<byte> request
                     _ = await sendSignal.WaitAsync(cancellationToken).ConfigureAwait(false);
                     try
                     {
+                        // If we bail while sending the PLC might still respond to the data that was sent.
+                        // This both breaks the send-one-receive-one flow as well as it might end up
+                        // completing a new job that reused the ID.
+#if NET5_0_OR_GREATER || NETSTANDARD2_1_OR_GREATER
+                        await
+#endif
+                        using var closeOnCancel = cancellationToken.MaybeUnsafeRegister(socket.Close);
```

Comment:
With the suggestion that this extension takes the state as argument too, this should be written as
```suggestion
                        using var closeOnCancel = cancellationToken.MaybeUnsafeRegister(static state =>
                        {
                            Socket socket = state as Socket;
                            socket.Close();
                        }, socket);
```
Thus (amortized) no allocaiton occurs.